bugs/PENDING_SUBMIT/0446-BAD-IMAGE-NVIDIA-0f62f70e-no-opt-test/stable-sampler-loop-min-texture-multiplied-mat4.amber

```diff
@@ -0,0 +1,407 @@
+         %61 = OpConstant %10 256
+         %70 = OpConstant %6 1
+         %72 = OpTypePointer Output %14
+         %73 = OpVariable %72 Output
+          %4 = OpFunction %2 None %3
+          %5 = OpLabel
+          %8 = OpVariable %7 Function
+         %13 = OpVariable %12 Function
+         %22 = OpVariable %21 Function
+               OpStore %8 %9
+         %17 = OpLoad %14 %16
+         %18 = OpVectorShuffle %11 %17 %17 0 1
+         %20 = OpVectorTimesScalar %11 %18 %19
+               OpStore %13 %20
+         %27 = OpLoad %24 %26
+         %28 = OpLoad %11 %13
+         %29 = OpImageSampleImplicitLod %14 %27 %28
+               OpStore %22 %29
+               OpBranch %30
+         %30 = OpLabel
+               OpLoopMerge %32 %33 None
+               OpBranch %34
+         %34 = OpLabel
+         %38 = OpAccessChain %37 %22 %36
+         %39 = OpLoad %10 %38
+         %41 = OpAccessChain %37 %22 %40
+         %42 = OpLoad %10 %41
+         %43 = OpFAdd %10 %39 %42
+         %45 = OpAccessChain %37 %22 %44
+         %46 = OpLoad %10 %45
+         %47 = OpFAdd %10 %43 %46
+         %50 = OpFOrdGreaterThan %49 %47 %48
+         %51 = OpLoad %6 %8
+         %53 = OpSLessThan %49 %51 %52
+         %54 = OpLogicalAnd %49 %50 %53
+               OpBranchConditional %54 %31 %32
+         %31 = OpLabel
+         %55 = OpLoad %14 %22
+         %56 = OpVectorShuffle %11 %55 %55 0 2
+         %57 = OpLoad %14 %22
+         %58 = OpVectorShuffle %11 %57 %57 1 1
+         %59 = OpFAdd %11 %56 %58
+               OpStore %13 %59
+         %60 = OpLoad %11 %13
+         %62 = OpVectorTimesScalar %11 %60 %61
+         %63 = OpExtInst %11 %1 Floor %62
+         %64 = OpCompositeConstruct %11 %61 %61
+         %65 = OpFDiv %11 %63 %64
+               OpStore %13 %65
+         %66 = OpLoad %24 %26
+         %67 = OpLoad %11 %13
+         %68 = OpImageSampleImplicitLod %14 %66 %67
+               OpStore %22 %68
+         %69 = OpLoad %6 %8
+         %71 = OpIAdd %6 %69 %70
+               OpStore %8 %71
+               OpBranch %33
+         %33 = OpLabel
+               OpBranch %30
+         %32 = OpLabel
+         %74 = OpLoad %14 %22
+               OpStore %73 %74
+               OpReturn
+               OpFunctionEnd
+END
+
+# uniforms for reference
+
+# tex
+SAMPLER reference_tex
+
+
+BUFFER reference_framebuffer FORMAT B8G8R8A8_UNORM
+
+PIPELINE graphics reference_pipeline
+  ATTACH reference_vertex_shader
+  ATTACH reference_fragment_shader
+  FRAMEBUFFER_SIZE 256 256
+  BIND BUFFER reference_framebuffer AS color LOCATION 0
+  BIND BUFFER default_texture AS combined_image_sampler SAMPLER reference_tex DESCRIPTOR_SET 0 BINDING 0
+END
+CLEAR_COLOR reference_pipeline 0 0 0 255
+
+CLEAR reference_pipeline
+RUN reference_pipeline DRAW_RECT POS 0 0 SIZE 256 256
+
+
+SHADER vertex variant_vertex_shader PASSTHROUGH
+
+# variant_fragment_shader is derived from the following GLSL:
+# #version 320 es
+#
+# precision highp int;
```

Comment:
Add this in the reference or remove it from here to make them even.

Reply:
The shader in question does the following:
- Take fragcoord to calculate a 1/256 granularity tex coordinates
- Fetch texel
- While sum of texel r+g+b values is over 1 (but only up to 16 times):
    - Calculate new texel coordinates based on previous texel coordinates
    - Normalize new coordinates to 1/256 granularity
    - Fetch new texel

The result is pseudorandom, but deterministic. Without digging any deeper, the only thing I can think of that might be a source of instability is possible differences in texture filtering. If the tex coordinates manage to get rounded by just one texel in any direction, the result should change. However, since the coordinates are normalized to 8 bits of precision, that feels unlikely.

If I was to debug this, I'd pick one broken output pixel, fix the fragcoord to those coordinates and then somehow output the values per iteration to see where things blow up.